common/src/main/java/xyz/wagyourtail/jsmacros/client/api/classes/Draw3D.java

```diff
@@ -604,12 +787,112 @@ public void render(MatrixStack matrixStack) {
             BufferBuilder buf = tess.getBuffer();
             Matrix4f model = matrixStack.peek().getPositionMatrix();
             RenderSystem.lineWidth(2.5F);
-            buf.begin(VertexFormat.DrawMode.DEBUG_LINES,  VertexFormats.POSITION_COLOR);
+            buf.begin(VertexFormat.DrawMode.DEBUG_LINES, VertexFormats.POSITION_COLOR);
             buf.vertex(model, (float) pos.x1, (float) pos.y1, (float) pos.z1).color(r, g, b, a).next();
             buf.vertex(model, (float) pos.x2, (float) pos.y2, (float) pos.z2).color(r, g, b, a).next();
             tess.draw();
-            
-            if (cull) RenderSystem.enableDepthTest();
+
+            if (cull) {
+                RenderSystem.enableDepthTest();
+            }
         }
+
     }
+
+    /**
+     * @since 1.6.5
+     */
+    public static class Surface extends Draw2D {
+        public final PositionCommon.Pos3D pos;
+        protected final PositionCommon.Pos3D rotations;
+        protected final PositionCommon.Pos2D sizes;
+        protected int minSubdivisions;
+
+        protected double scale;
+        public boolean renderBack;
+        public boolean cull;
+
+        public Surface(PositionCommon.Pos3D pos, PositionCommon.Pos3D rotations, PositionCommon.Pos2D sizes, int minSubdivisions, boolean renderBack, boolean cull) {
+            this.pos = pos;
+            this.rotations = rotations;
+            this.sizes = sizes;
+            this.minSubdivisions = minSubdivisions;
+            this.renderBack = renderBack;
+            this.cull = cull;
+            init();
+        }
+
+        public void setPos(double x, double y, double z) {
+            this.pos.x = x;
+            this.pos.y = y;
+            this.pos.z = z;
+            init();
+        }
+
```

Comment:
No setters for rotations and sizes?

```suggestion

        public void setRotations(double xRot, double yRot, double zRot) {
            this.rotations.x = xRot;
            this.rotations.y = yRot;
            this.rotations.z = zRot;
            init();
        }

        public void setSizes(double width, double height) {
            this.sizes.x = width;
            this.sizes.y = height;
            init();
        }

```